azurerm/internal/services/sql/resource_arm_sql_database.go

```diff
@@ -449,6 +455,10 @@ func resourceArmSqlDatabaseCreateUpdate(d *schema.ResourceData, meta interface{}
 		properties.DatabaseProperties.ReadScale = sql.ReadScaleDisabled
 	}
 
+	if v, ok := d.GetOkExists("zone_redundant"); ok {
+		properties.DatabaseProperties.ZoneRedundant = utils.Bool(v.(bool))
```

Comment:
since this field'll no longer be computed we can assign this in the `CreateProperties` block above